c10/util/thread_name.cpp

```diff
@@ -19,8 +19,8 @@ namespace c10 {
 
 void setThreadName(std::string name) {
 #ifdef C10_HAS_PTHREAD_SETNAME_NP
-  constexpr size_t kMaxThreadName = 15;
-  name.resize(std::min(name.size(), kMaxThreadName));
+  constexpr size_t kMaxThreadName = 16;
+  name.resize(std::min(name.size(), kMaxThreadName), '\0');
```

Comment:
Isn't this only going to pad with `\0` but not add it at the end if the string is too long?
That's how I read the old size being 15, when you get the c_str(), it will always add the `\0` after it making it 16.

Reply:
@albanD I reverted it to the original implementation. I realized that I had used a longer name in my tests.